File: guides/java/kora-java-guide-dependency-injection/kora-java-guide-dependency-injection-app/src/main/java/io/koraframework/guide/dependencyinjection/activity/ActivityService.java
package io.koraframework.guide.dependencyinjection.activity;

import io.koraframework.application.graph.ValueOf;
import io.koraframework.common.annotation.Component;

@Component
public final class ActivityService {

    private final ValueOf<ActivityRecorder> activityRecorder;

    public ActivityService(ValueOf<ActivityRecorder> activityRecorder) {
        this.activityRecorder = activityRecorder;
        System.out.println("ActivityService created (ActivityRecorder not yet accessed)");
    }

    public void recordActivityByUserName(String user) {
        System.out.println("Recording activity for: " + user);
        ActivityRecorder recorder = activityRecorder.get();
        recorder.recordUser(user);
        System.out.println("Activity recorded successfully");
    }
}

File: guides/kotlin/kora-kotlin-guide-dependency-injection/kora-kotlin-guide-dependency-injection-app/src/main/kotlin/io/koraframework/guide/dependencyinjection/activity/ActivityService.kt
package io.koraframework.guide.dependencyinjection.activity

import io.koraframework.application.graph.ValueOf
import io.koraframework.common.annotation.Component

@Component
class ActivityService(
    private val activityRecorder: ValueOf<ActivityRecorder>
) {

    init {
        println("ActivityService created (ActivityRecorder not yet accessed)")
    }

    fun recordActivityByUserName(user: String) {
        println("Recording activity for: $user")
        val recorder = activityRecorder.get()
        recorder.recordUser(user)
        println("Activity recorded successfully")
    }
}
